back-end/controllers/busCapacityController.js

```diff
@@ -0,0 +1,86 @@
+const BusCapacity = require("../model/bus-capacity");
+const isNil = require("../utils");
+const Interval_limit = 30;
+
+const getBusCapacity = async (req, res) => {
+  // remove outdated capacity info
+  let oldBusCapacity = await BusCapacity.aggregate([
+    {
+      $addFields: {
+        reportTimeDiff: {
+          $dateDiff: {
+            startDate: "$ReportTime",
+            endDate: "$$NOW",
+            unit: "minute",
+          },
+        },
+      },
+    },
+    {
+      $match: { reportTimeDiff: { $gt: Interval_limit } },
+    },
+  ]);
+  oldBusCapacity = oldBusCapacity.map(function (bus) {
+    return bus.VehicleNo;
+  });
+  await BusCapacity.deleteMany({ VehicleNo: { $in: oldBusCapacity } });
```

Comment:
I suggest that we move this into a cron task - I can set it up later. That should speed up this function a bit since we don't need to clear the outdated capacities each time they're fetched, maybe only once every few minutes